### DISCUSSION TITLE
How to mount cluster/repo secrets directly to argo-cd server

### DISCUSSION THREAD

**Discussion by CRASH-Tech** [argoproj/argo-cd#14140] [Q&A]
Hello, I want to inject/mount clusters/repo secrets directly to argo-cd via vault injector. And don't keep it in the kubernetes secrets, because kubernetes secrets are insecure storage.
How can I do it?

**Comment by crenshaw-dev** [comment]
I don't think Kubernetes Secrets are inherently insecure. They just have to be secured. See, for example: https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/

**Reply by CRASH-Tech** [reply]
But in-memory secrets are more secured

**Reply by crenshaw-dev** [reply]
If you feel super strongly about it, you could contribute a feature to load those secrets from the FS. But my feeling is that Secrets are a well-established and maintained feature of Kubernetes and that the effort is better spent hardening Kubernetes.
